core/common/src/ClipStyle.ts

```diff
@@ -128,39 +229,49 @@ export class ClipStyle {
    * @note Cut geometry will only be produced for element geometry - not for, e.g., terrain or reality models.
    */
   public readonly produceCutGeometry: boolean;
+  /** If 'true', intersection of geometry and clip planes will be colorized */
+  public readonly colorizeIntersection: boolean;
   /** Controls aspects of how the cut geometry is displayed, if [[produceCutGeometry]] is `true`. */
   public readonly cutStyle: CutStyle;
   /** If defined, geometry inside the clip planes will be drawn in this color. */
   public readonly insideColor?: RgbColor;
   /** If defined, geometry outside of the clip planes will be drawn in this color instead of being clipped. */
   public readonly outsideColor?: RgbColor;
+  /** Controls the style of the intersection of geometry and clip planes */
+  public readonly intersectionStyle?: ClipIntersectionStyle;
 
   /** The default style, which overrides none of the view's settings. */
-  public static readonly defaults = new ClipStyle(false, CutStyle.defaults, undefined, undefined);
+  public static readonly defaults = new ClipStyle(false, false, CutStyle.defaults, undefined, undefined, undefined);
 
-  private constructor(produceCutGeometry: boolean, cutStyle: CutStyle, inside: RgbColor | undefined, outside: RgbColor | undefined) {
+  private constructor(produceCutGeometry: boolean, colorizeIntersection: boolean, cutStyle: CutStyle, inside: RgbColor | undefined, outside: RgbColor | undefined, intersectionStyle: ClipIntersectionStyle | undefined) {
     this.produceCutGeometry = produceCutGeometry;
+    this.colorizeIntersection = colorizeIntersection;
     this.cutStyle = cutStyle;
     this.insideColor = inside;
     this.outsideColor = outside;
+    this.intersectionStyle = intersectionStyle;
   }
 
   /** Create a style from its components. */
-  public static create(produceCutGeometry: boolean, cutStyle: CutStyle, insideColor?: RgbColor, outsideColor?: RgbColor): ClipStyle {
-    if (!produceCutGeometry && cutStyle.matchesDefaults && !insideColor && !outsideColor)
+  // public static create(produceCutGeometry: boolean, cutStyle: CutStyle, colorizeIntersection: boolean, insideColor?: RgbColor, outsideColor?: RgbColor, intersectionStyle?: ClipIntersectionStyle): ClipStyle;
+
+  public static create(style: ClipStyleArgs): ClipStyle {
```

Comment:
You have changed the signature of a `@public` method. That's [not permitted](https://github.com/iTwin/itwinjs-core/blob/master/docs/learning/api-support-policies.md). Deprecate the original method (with none of your additional arguments) and add a new overload that takes `ClipStyleArgs`.

Reply:
TypeScript cannot overload same-named functions with differing amounts of parameters.
The old create() took a list of individual parameters.
Andre has changed that to take a single arguments object containing properties for each argument.
It seems we might need to do the following:
- deprecate create(); in comment tell user to call a new createClipStyleFromArgs method.
- make a new method createClipStyleFromArgs which takes a the argument object.

Does this seem the right approach?

Reply:
> TypeScript cannot overload same-named functions with differing amounts of parameters.

[Yes, it can](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAQwM6oKYCcoAoIIAmMsCAXIgEZxwA2GyYAlBWpjqogWMaUjJyhYQGRAG8AUImmIYwRLgCE3XvGZSZmqAAsscAO6IwGQwFEserLgBEAQXTY+iYMhi0QWDNaYBuCQF8JCRgwKGwXCFFbLABzTkktbRCYijAQAFtKbD9NVCgQYGAKPKxkv0CJUEgndIBPADFwCFxkWNQKaLiWRBKyyqaahqbcHWTUjKysABoe-MLioWTu3rAYvwB6ACpNxAABELCsMGRaRE31-uq1RDrGyBGk1YB5LE72ownsRAAfRDeZvIFYAAfgWpVWTHEGmk9CgiFGqxyMjkCigtQADhg4PIETEXm9EIoALxExDWNKZbDeKGaRLJRCk3H4toAOlxSNpsyBDPhjzxr1ZgMKHMQ-kQGFomBpnJQDhwuHAhAwwBCGEIhJJXMKvmhtNxPKZAriIoqmk8+SOiAABgASMS4-wzO1C4D+K3lILcVB0DAs2hwGK4W7DMS8saIACMALmRTJCC8oqYOq9Pr9AaDQ3uACYZtYoPo4N5fEA).

Reply:
> > TypeScript cannot overload same-named functions with differing amounts of parameters.
> 
> [Yes, it can](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAQwM6oKYCcoAoIIAmMsCAXIgEZxwA2GyYAlBWpjqogWMaUjJyhYQGRAG8AUImmIYwRLgCE3XvGZSZmqAAsscAO6IwGQwFEserLgBEAQXTY+iYMhi0QWDNaYBuCQF8JCRgwKGwXCFFbLABzTkktbRCYijAQAFtKbD9NVCgQYGAKPKxkv0CJUEgndIBPADFwCFxkWNQKaLiWRBKyyqaahqbcHWTUjKysABoe-MLioWTu3rAYvwB6ACpNxAABELCsMGRaRE31-uq1RDrGyBGk1YB5LE72ownsRAAfRDeZvIFYAAfgWpVWTHEGmk9CgiFGqxyMjkCigtQADhg4PIETEXm9EIoALxExDWNKZbDeKGaRLJRCk3H4toAOlxSNpsyBDPhjzxr1ZgMKHMQ-kQGFomBpnJQDhwuHAhAwwBCGEIhJJXMKvmhtNxPKZAriIoqmk8+SOiAABgASMS4-wzO1C4D+K3lILcVB0DAs2hwGK4W7DMS8saIACMALmRTJCC8oqYOq9Pr9AaDQ3uACYZtYoPo4N5fEA).

Ah, ok, thanks for the sample.  @AJMigliore will give it a try.

Reply:
This has been addressed with the most recent changes.